python-package/lightgbm/sklearn.py

```diff
@@ -529,6 +529,9 @@ def _more_tags(self):
             }
         }
 
+    def __sklearn_is_fitted__(self) -> bool:
+        return getattr(self, "fitted_", False)
```

Comment:
We set `fitted_` attribute at the end of the `fit()` method
https://github.com/microsoft/LightGBM/blob/a77260f03eb735923125024b0f36df9f63a6e0f3/python-package/lightgbm/sklearn.py#L770

Reply:
I noticed in the https://github.com/dmlc/xgboost/pull/7230, they also replaced some other internal checks of estimator attributes with this method call.

I think that's a good idea, so only this method needs to know about the `fitted_` property. What do you think?

That would mean changing the following to `if self.__sklearn_is_fitted__()`:

https://github.com/microsoft/LightGBM/blob/a77260f03eb735923125024b0f36df9f63a6e0f3/python-package/lightgbm/dask.py#L1004-L1005

Reply:
I agree. I remember you were asking why do we use `if self._n_features is None` everywhere for checking that estimator is fitted when we have `self.fitted_`.
https://github.com/microsoft/LightGBM/blob/a77260f03eb735923125024b0f36df9f63a6e0f3/python-package/lightgbm/sklearn.py#L857
https://github.com/microsoft/LightGBM/pull/3883#discussion_r568859996
My answer was that it's because `self.fitted_` was introduced much later. Now we can replace everything with `self.__sklearn_is_fitted__()`. And I think it will be clearer.

But I'd prefer to do it in a follow-up PR, if you do not mind.

Reply:
ok yep, I'm fine with it being a followup